File: apps/ios/Sources/Core/Navigation/ForegroundState.swift
struct ForegroundState: Sendable {
    private(set) var active = false
    private(set) var epoch: UInt64 = 0
    mutating func transition(_ value: Bool) {
        guard active != value else { return }
        active = value
        if value { epoch += 1 }
    }
}

File: apps/android/app/src/main/java/chat/rogi/rogichat/core/navigation/ForegroundState.kt
package chat.rogi.rogichat.core.navigation

// One app-shell source. Repeated active callbacks do not create duplicate refresh epochs.
data class ForegroundState(val active: Boolean = false, val epoch: Long = 0) {
    fun transition(value: Boolean) = if (value == active) this
        else copy(active = value, epoch = if (value) epoch + 1 else epoch)
}
